src/collector/polymer/polymer.worker.ts

```diff
@@ -246,43 +249,20 @@ class PolymerCollectorSnifferWorker {
     ): Promise<void> {
 
         const event = parsedLog.args as unknown as SendPacketEvent.OutputObject;
-        const destinationChain: string = event.sourceChannelId;
+
+        // We need to convert the source chain id to the destination chain using the
+        // settings map.
+        this.config.polymerChannels[event.sourceChannelId];
+        const destinationChain: string = this.config.polymerChannels[event.sourceChannelId] ?? event.sourceChannelId;
```

Comment:
If the mapping of `sourceChannelId` within `polymerChannels` is not found I'd simply return early and not continue to process the event. Having an 'unknown' channel id as the `destinationChain` might cause problems further along the line.